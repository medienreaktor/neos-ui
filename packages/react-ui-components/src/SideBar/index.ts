import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from '../identifiers';

import SideBar from './sideBar';
import style from './style.css';

export default themr(identifiers.sideBar, style)(SideBar);
