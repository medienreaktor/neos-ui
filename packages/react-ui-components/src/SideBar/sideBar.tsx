import mergeClassNames from 'classnames';
import React, {PureComponent} from 'react';

type SideBarPosition = 'left' |Â 'right';

interface SideBarTheme {
    readonly 'sideBar': string;
    readonly 'sideBar--left': string;
    readonly 'sideBar--right': string;
}

export interface SideBarProps {
    /**
     * This prop controls the absolute positioning of the SideBar.
     */
    readonly position: SideBarPosition;

    /**
     * An optional className to render on the div node.
     */
    readonly className?: string;

    /**
     * The children to render within the div node.
     */
    readonly children: React.ReactNode;

    /**
     * An optional css theme to be injected.
     */
    readonly theme?: SideBarTheme;
}

class SideBar extends PureComponent<SideBarProps> {
    public render(): JSX.Element {
        const {
            position,
            className,
            children,
            theme,
            ...rest
        } = this.props;
        const classNames = mergeClassNames(
            className,
            theme!.sideBar,
            {
                [theme!['sideBar--left']]: position === 'left',
                [theme!['sideBar--right']]: position === 'right'
            }
        );

        return (
            <div {...rest} className={classNames}>
                {children}
            </div>
        );
    }
}

export default SideBar;
