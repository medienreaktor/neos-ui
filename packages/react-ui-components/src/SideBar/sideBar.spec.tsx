import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import SideBar, {SideBarProps} from './sideBar';

describe('<SideBar/>', () => {
    const props: SideBarProps = {
        children: 'Foo children',
        position: 'left',
        theme: {
            'sideBar': 'sideBarClassName',
            'sideBar--left': 'leftClassName',
            'sideBar--right': 'rightClassName',
        }
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<SideBar {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<SideBar {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<SideBar {...props} aria-hidden={true}/>);
        expect(wrapper.prop('aria-hidden')).toBe(true);
    });
});
