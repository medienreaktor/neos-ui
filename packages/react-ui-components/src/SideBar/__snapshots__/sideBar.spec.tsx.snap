// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar/> should render correctly. 1`] = `
<div
  className="sideBarClassName leftClassName"
>
  Foo children
</div>
`;
